packages/wrangler/src/type-generation.ts

```diff
@@ -0,0 +1,171 @@
+import * as fs from "fs";
+import { findUpSync } from "find-up";
+import { getEntry } from "./entry";
+import { logger } from "./logger";
+import type { Config } from "./config";
+import type { CfWorkerInit } from "./worker";
+
+// Currently includes bindings & rules for declaring modules
+export type PartialConfigToDTS = CfWorkerInit["bindings"] & {
+	rules: Config["rules"];
+};
+export async function generateTypes(
+	configToDTS: PartialConfigToDTS,
+	config: Config
+) {
+	const entry = await getEntry({}, config, "types");
+	const envTypeStructure: string[] = [];
+
+	if (configToDTS.kv_namespaces) {
```

Comment:
If you give the config default values for `kv_namespaces` etc... of an empty array, you could remove all these ifs

Reply:
I will consider it.